zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java

```diff
@@ -420,8 +435,10 @@ private void updateParagraph(NotebookSocket conn, Notebook notebook,
     Paragraph p = note.getParagraph(paragraphId);
     p.settings.setParams(params);
     p.setConfig(config);
-    p.setTitle((String) fromMessage.get("title"));
-    p.setText((String) fromMessage.get("paragraph"));
+    if (!conf.getBoolean(ConfVars.ZEPPELIN_READ_ONLY)) {
```

Comment:
@Xrixcis  I think it is intended to make Zeppelin readonly(blocks notebook/paragraph CRUD), but let user still able to run.